src/main/java/com/newrelic/opentracing/aws/TracingRequestHandler.java

```diff
@@ -42,16 +43,18 @@ default Output handleRequest(Input input, Context context) {
     final Tracer tracer = GlobalTracer.get();
     final SpanContext spanContext = extractContext(tracer, input);
 
-    try (Scope scope = tracer.buildSpan("handleRequest").asChildOf(spanContext).startActive(true)) {
-      SpanUtil.setTags(scope, context, input, isColdStart);
+    Span span = SpanUtil.buildSpan(input, context, tracer, spanContext);
+    try (Scope scope = tracer.activateSpan(span)) {
       try {
         Output output = doHandleRequest(input, context);
-        ResponseParser.parseResponse(output, scope.span());
+        ResponseParser.parseResponse(output, span);
         return output;
       } catch (Throwable throwable) {
-        scope.span().log(SpanUtil.createErrorAttributes(throwable));
+        span.log(SpanUtil.createErrorAttributes(throwable));
         throw throwable;
       }
+    } finally {
+      span.finish();
```

Comment:
am I the only one annoyed that `Span` doesn't implement `Closeable`, so you could _really_ use try-with-resources?

Reply:
yeah, the choice made to move away from making the Span Auto-Closable was a surprise to me.